### PR TITLE
[feature]:增加用户登录接口，增加修饰器模块

### PR DESCRIPTION
**Pull request by Maryin-c**
*Issue #, if available:*

*Description of changes:*
增加用户登录接口
在wrappers.py中添加view.login相关装饰器函数，修复redis的部分bug

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: dkhtn_django/user/tests.py [hunk 1]
```diff
@@ -54,7 +53,43 @@ def test_email_send(client, get_url, status_code, info_dict):
...
+    response = client.get(url)
+    assert response.status_code == status_code
+    assert response.json() == info_dict
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "url, info, status_code, info_dict",
+    [
+        # (
```
Comment (Carkham) [inline dkhtn_django/user/tests.py:66]: 注释删了

File: dkhtn_django/user/views.py [hunk 2]
```diff
@@ -33,3 +37,35 @@ def rsa_get(request):
...
+            return JsonResponse(response)
+        else:
+            response = {
+                "code": 0,
+                "message": "登录成功",
+            }
+            return JsonResponse(response)
+    except Exception as e:
+        raise e
+        # response = {
```
Comment (Carkham) [inline dkhtn_django/user/views.py:67]: 这又在注释什么

File: dkhtn_django/user/wrappers.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    redis_utils.redis_set(settings.REDIS_DB_LOGIN, request.userinfo['id'], new_session_id)
+
+
+# login接口专用，设置为无条件登录，并且拒绝多点登录
+# 维持登陆状态的redis映射：session_id->{id, name, avatar, email}
+# 检测多点登录的redis映射：id->session_id
+def wrapper_set_login(func):
+    def inner(request, *args, **kwargs):
+        # 在调用view函数前执行
+        pass
```
Comment (Carkham) [inline dkhtn_django/user/wrappers.py:44]: 这个pass是什么意思

File: dkhtn_django/utils/redis_utils.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from django_redis import get_redis_connection
+from django.conf import settings
+
+redis_clis = {
+    'default': get_redis_connection('default'),
+    '1': get_redis_connection('1'),
+}
+
+
+def redis_get(db_index, key):
```
Comment (Carkham) [inline dkhtn_django/utils/redis_utils.py:10]: 加一个连接检查装饰器

**Review by Carkham — COMMENTED** [review]
小修

**Review by Carkham — COMMENTED** [review]
dockformat

**Review by Xbuluo — APPROVED** [review]

**Pull request closed without merging**